### PR TITLE
libguestfs: 1.48.0 -> 1.48.4

### PR DESCRIPTION
**Pull request by twz123**
###### Description of changes

* Fix Perl dependencies, so that the Perl bindings will be built
* Move OCaml bindings into the proper place, so they can be picked up

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by symphorien** [comment]
`virt-tar-in --help` fails because of the wrapper. could you fix this? This is not a regression so it can go in a later PR if you prefer.

otherwise LGTM

File: pkgs/development/libraries/libguestfs/default.nix [hunk 1]
```diff
@@ -1,37 +1,95 @@
...
+, libapparmor
+, jansson
+, getopt
+, perlPackages
+, ocamlPackages
 , libtirpc
 , appliance ? null
-, javaSupport ? false, jdk ? null }:
+, javaSupport ? false
+, jdk ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libguestfs/default.nix:39]:
```suggestion
, jdk
```

File: pkgs/development/libraries/libguestfs/default.nix [hunk 2]
```diff
@@ -1,37 +1,95 @@
...
+}:
 
 assert appliance == null || lib.isDerivation appliance;
 assert javaSupport -> jdk != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libguestfs/default.nix:43]:
```suggestion
```

File: pkgs/development/libraries/libguestfs/default.nix [hunk 3]
```diff
@@ -54,10 +112,14 @@ stdenv.mkDerivation rec {
...
     ./libguestfs-syms.patch
   ];
+
+  createFindlibDestdir = true;
+
   installFlags = [ "REALLY_INSTALL=yes" ];
   enableParallelBuilding = true;
 
   postInstall = ''
+    mv -- "$out/lib/ocaml/guestfs" "$OCAMLFIND_DESTDIR/guestfs"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libguestfs/default.nix:122]:
```suggestion
    mv "$out/lib/ocaml/guestfs" "$OCAMLFIND_DESTDIR/guestfs"
```

**Comment by twz123** [comment]
> `virt-tar-in --help` fails because of the wrapper. could you fix this? This is not a regression so it can go in a later PR if you prefer.
> 
> otherwise LGTM

I'm not sure that the wrappers are even needed any more. All of the bins are compiled binaries or small shell wrappers, so I guess they don't need the Perl libs, but I don't know if they shell out to qemu.

In case they are still needed: I reckon the `exec -a "$0"` pattern doesn't work when invoking other scripts. How would I get the env vars into those virt-* scripts? some `substituteInPlace` trickery?

**Comment by SuperSandro2000** [comment]
> In case they are still needed: I reckon the `exec -a "$0"` pattern doesn't work when invoking other scripts. How would I get the env vars into those virt-* scripts? some `substituteInPlace` trickery?

You can wrap any script or executable and set extra environment variables.

**Comment by twz123** [comment]
> You can wrap any script or executable and set extra environment variables.

By "wrap" you mean `wrapProgram`, right? This actually breaks the virt-* scripts, as they rely on a correct `$0` (which apparently can't be passed via `exec -a` to scripts, but just to binaries):

https://github.com/libguestfs/libguestfs/blob/fadce8d3e7cffd4eadc982e330285a93a3486368/fish/virt-tar-in#L19-L27

**Comment by symphorien** [comment]
Yes wrapping by a shell script interferes with $0. you can use makeBinaryWrapper instead to solve it (https://nixos.org/manual/nixpkgs/unstable/#fun-makeWrapper)

**Comment by twz123** [comment]
I exchanged makeWrapper with makeBinaryWrapper, but it still doesn't seem to work...

```diff
diff --git a/pkgs/development/libraries/libguestfs/default.nix b/pkgs/development/libraries/libguestfs/default.nix
index e2da1a30ff1..cbe01427ef9 100644
--- a/pkgs/development/libraries/libguestfs/default.nix
+++ b/pkgs/development/libraries/libguestfs/default.nix
@@ -3,7 +3,7 @@
 , fetchurl
 , pkg-config
 , autoreconfHook
-, makeWrapper
+, makeBinaryWrapper
 , ncurses
 , cpio
 , gperf
@@ -59,7 +59,7 @@ stdenv.mkDerivation rec {
     flex
     getopt
     gperf
-    makeWrapper
+    makeBinaryWrapper
     pkg-config
     qemu
   ] ++ (with perlPackages; [ perl libintl-perl GetoptLong ModuleBuild ])

```

```console
❯ for x in /nix/store/ihl7d46wi85jyyd9b0ahlw43qaazslmd-libguestfs-1.48.4/bin/*; do y=$(LC_ALL=C "$x" --help 2>&1) || echo "$x: $y" ; done
/nix/store/ihl7d46wi85jyyd9b0ahlw43qaazslmd-libguestfs-1.48.4/bin/virt-copy-in: No manual entry for .virt-copy-in-wrapped
/nix/store/ihl7d46wi85jyyd9b0ahlw43qaazslmd-libguestfs-1.48.4/bin/virt-copy-out: No manual entry for .virt-copy-out-wrapped
/nix/store/ihl7d46wi85jyyd9b0ahlw43qaazslmd-libguestfs-1.48.4/bin/virt-tar-in: No manual entry for .virt-tar-in-wrapped
/nix/store/ihl7d46wi85jyyd9b0ahlw43qaazslmd-libguestfs-1.48.4/bin/virt-tar-out: No manual entry for .virt-tar-out-wrapped
```

```
# ------------------------------------------------------------------------------------
# The C-code for this binary wrapper has been generated using the following command:


makeCWrapper '/nix/store/ihl7d46wi85jyyd9b0ahlw43qaazslmd-libguestfs-1.48.4/bin/.virt-copy-in-wrapped' \
    --inherit-argv0 \
    --prefix 'PATH' ':' '/nix/store/ihl7d46wi85jyyd9b0ahlw43qaazslmd-libguestfs-1.48.4/bin:/nix/store/8pr3v5h8b6qzy49hdwcl2d2pi014h771-hivex-1.3.21/bin:/nix/store/j0zcx96skkssw0bhmfy93dgamqg5bg1w-qemu-7.0.0/bin' \
    --prefix 'PERL5LIB' ':' '/nix/store/ihl7d46wi85jyyd9b0ahlw43qaazslmd-libguestfs-1.48.4/lib/perl5/site_perl'


# (Use `nix-shell -p makeBinaryWrapper` to get access to makeCWrapper in your shell)
# ------------------------------------------------------------------------------------
```

So, if you don't mind I'd leave this issue for another PR.

**Comment by symphorien** [comment]
>  Oh sure, the issue is with scripts in general, so it doesn't help much that the wrapper is a binary. The issue is that the wrapped file itself is a script, right?

I'm not sure honestly.

**Pull request merged**